[GENERAL UPDATE] Add variables $VERSION to agility the installing and updating process of the services

Swissnode opened the pull request:
#### What

Added two variables $VERSION so that users upgrading to a version that is not already mirrored in the documentation only have to change the VERSION variable once and can merely copy-paste the other lines

### Why

Helps users to upgrade quicker. Very simple change too!

#### How

Code change to add the two variables mentioned above

#### Scope

- [ ] significant change to core configuration
- [ ] independent bonus guide
- [ ] simple bug fix

Fixes # (link issue)
N/A - not a fix

#### Test & maintenance

Can be tested by merely updating a future lnd upgrade using it

kristapsk commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,21 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-v0.15.5-beta.txt
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
```
It creates variable and assigns value to it. Works the same way in interactive bash shell and bash scripts.

kristapsk commented:
Concept ACK. Will be able to look at it more carefully and test in a few days.

vicariousdrama commented on guide/lightning/lightning-client.md:
````diff
@@ -63,7 +65,7 @@ Now that we've verified the integrity of the downloaded binary, we need to check
 
   ```sh
-  $ gpg --verify manifest-roasbeef-v0.15.5-beta.sig manifest-v0.15.5-beta.txt
+  $ gpg --verify manifest-roasbeef-$VERSION.sig manifest-v0.15.5-beta.txt
````
Need the $VERSION reference to be used in the manifest filename as well.

Mynima replied:
@Swissnode this minor update still outstanding

Swissnode replied:
oh shit sorry, didn't see that, as I hardly ever log on to Github.

Will get on it...

kristapsk commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,21 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
+  $ DISTRO="lnd-linux-arm64"
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/$DISTRO-$VERSION.tar.gz
```
I think this makes more sense.
```suggestion
  $ DISTRO="linux-arm64"
  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/lnd-$DISTRO-$VERSION.tar.gz
```

twofaktor replied:
DISTRO variable is not necessary, RaspiBolt will be always for arm64

kristapsk commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,21 @@ We'll download, verify and install LND.
-  $ sha256sum --check manifest-v0.15.5-beta.txt --ignore-missing
-  > lnd-linux-arm64-v0.15.5-beta.tar.gz: OK
+  $ sha256sum --check manifest-$VERSION.txt --ignore-missing
+  > lnd-linux-arm64-$VERSION.tar.gz: OK
```
You will not have `$VERSION` but actual version in output here.
```suggestion
  > lnd-linux-arm64-v0.15.5-beta.tar.gz: OK
```

Mynima replied:
@Swissnode this minor update still outstanding

Swissnode replied:
yeah i didn't realise... on it... (my first ever PR)

Swissnode replied:
I think this change is taken care of in a later suggestion... (was getting conflicting lines) - i think we can ignore this suggestion from December...

twofaktor commented:
Good add-on! This will go to agility the updating process of the services a lot. BTW: Could apply this to other services installation e.g to the Bitcoin client as well?

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,21 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-v0.15.5-beta.txt
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
```
Should we call this `LNDVERSION="v0.15.5-beta"` to can use different variables for each service in the future?

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,21 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
+  $ DISTRO="lnd-linux-arm64"
```
Should we call this `LNDDISTRO="v0.15.5-beta"` to can use different variables for each service in the future? 

i.e: BTCVERSION / BTCDISTRO

Mynima commented:
> Good add-on! This will go to agility the updating process of the services a lot. BTW: Could apply this to other services installation e.g to the Bitcoin client as well?

yeah I was wondering the same thing. It would be great to add this same methodology with the other sections. Def think Bitcoin Client is a good place to start.

Mynima reviewed (commented):
ACK Couple of minor comments to be resolved. Have tested throughout without issue.

twofaktor commented:
> $ BVERSION="24.0.1" 

Why "BVERSION" and not "VERSION" ??

Mynima commented:
> > $ BVERSION="24.0.1"
> 
> Why "BVERSION" and not "VERSION" ??

Absolutely no reason to not be VERSION and probably should be for consistency.

twofaktor commented:
> > > $ BVERSION="24.0.1"
> > 
> > 
> > Why "BVERSION" and not "VERSION" ??
> 
> Absolutely no reason to not be VERSION and probably should be for consistency.

Environment variables are temporal definitions, these only keep in an current session

twofaktor commented:
@Swissnode are you available to do modifications? do you prefer we close this PR and I open a new one so I can take charge

Mynima commented:
yo @Swissnode would you have time to update this one for the new LND release [lnd v0.16.1-beta](https://github.com/lightningnetwork/lnd/releases/tag/v0.16.1-beta), I'm happy to do the testing when I upgrade at the weekend. Be great to get this PR closed out.

Swissnode commented:
Hey guys,

Really sorry for the delay - I will try and get on my account more often.

I implemented the changes in my fork. Can you have a look?

@twofaktor thanks

Mynima commented:
> Could we add Bitcoin Core and other possible services to this PR too?
> 
> I already integrated that on MiniBolt guide, we could get references of there if you agree

Yeah I think this could be a good idea to get them done at the same time. Will you be able to do that @Swissnode?

Swissnode commented:
> > Could we add Bitcoin Core and other possible services to this PR too?
> > I already integrated that on MiniBolt guide, we could get references of there if you agree
> 
> Yeah I think this could be a good idea to get them done at the same time. Will you be able to do that @Swissnode?

Yeah I guess I can do that!!

Excuse my Github noobishness on this front. I assume I go into my own SwissNode repository and do the changes there correct? Do I somehow have to tag WHICH files are included or will Github assume that ALL changes i make to my repository will be taken into account for this PR?

Swissnode commented:
ok @Mynima and @twofaktor 

I see it does pick up all the files I changed so I am now fairly sure I got the changes correctly made. Wanna review and pull them in?

Updated bitcoin-client.md and lightning-client.md

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -44,14 +44,18 @@ This is a precaution to make sure that this is an official release and not a mal
+  
   # download Bitcoin Core binary
-  $ wget https://bitcoincore.org/bin/bitcoin-core-23.0/bitcoin-23.0-aarch64-linux-gnu.tar.gz
+  $ wget https://bitcoincore.org/bin/$VERSION_CORE/$VERSION-aarch64-linux-gnu.tar.gz
```
```suggestion
  $ wget https://bitcoincore.org/bin/bitcoin-core-$VERSION/bitcoin-$VERSION-aarch64-linux-gnu.tar.gz
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
````diff
@@ -44,14 +44,18 @@ This is a precaution to make sure that this is an official release and not a mal
   ```sh
+  # set up some version variables for easier maintenance later on
+  $ VERSION_CORE="bitcoin-core-25.0"
+  $ VERSION="bitcoin-25.0"
````
```suggestion
  $ VERSION="25.0"
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -44,14 +44,18 @@ This is a precaution to make sure that this is an official release and not a mal
 
   # download the list of cryptographic checksum
-  $ wget https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS
+  $ wget https://bitcoincore.org/bin/$VERSION_CORE/SHA256SUMS
```
```suggestion
  $ wget https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -44,14 +44,18 @@ This is a precaution to make sure that this is an official release and not a mal
 
   # download the signatures attesting to validity of the checksums
-  $ wget https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS.asc
+  $ wget https://bitcoincore.org/bin/$VERSION_CORE/SHA256SUMS.asc
```
```suggestion
  $ wget https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS.asc
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -91,8 +95,8 @@ This is a precaution to make sure that this is an official release and not a mal
 * The binary checksum file is timestamped on the Bitcoin blockchain via the [OpenTimestamps protocol](https://opentimestamps.org/){:target="_blank"}, proving that the file existed prior to some point in time. Let's verify this timestamp. On your local computer, download the checksums file and its timestamp proof:
-  *  https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS.ots
-  *  https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS
+  *  https://bitcoincore.org/bin/bitcoin-25.0/SHA256SUMS.ots
```
```suggestion
  *  https://bitcoincore.org/bin/bitcoin-core-25.0/SHA256SUMS.ots
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -91,8 +95,8 @@ This is a precaution to make sure that this is an official release and not a mal
-  *  https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS.ots
-  *  https://bitcoincore.org/bin/bitcoin-core-23.0/SHA256SUMS
+  *  https://bitcoincore.org/bin/bitcoin-25.0/SHA256SUMS.ots
+  *  https://bitcoincore.org/bin/bitcoin-25.0/SHA256SUMS
```
```suggestion
  *  https://bitcoincore.org/bin/bitcoin-core-25.0/SHA256SUMS
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
````diff
@@ -105,10 +109,10 @@ This is a precaution to make sure that this is an official release and not a mal
   ```sh
-  $ tar -xvf bitcoin-23.0-aarch64-linux-gnu.tar.gz
-  $ sudo install -m 0755 -o root -g root -t /usr/local/bin bitcoin-23.0/bin/*
+  $ tar -xvf $VERSION-aarch64-linux-gnu.tar.gz
````
```suggestion
  $ tar -xvf bitcoin-$VERSION-aarch64-linux-gnu.tar.gz
```

twofaktor commented on guide/bitcoin/bitcoin-client.md:
```diff
@@ -105,10 +109,10 @@ This is a precaution to make sure that this is an official release and not a mal
-  $ tar -xvf bitcoin-23.0-aarch64-linux-gnu.tar.gz
-  $ sudo install -m 0755 -o root -g root -t /usr/local/bin bitcoin-23.0/bin/*
+  $ tar -xvf $VERSION-aarch64-linux-gnu.tar.gz
+  $ sudo install -m 0755 -o root -g root -t /usr/local/bin $VERSION/bin/*
```
```suggestion
  $ sudo install -m 0755 -o root -g root -t /usr/local/bin bitcoin-$VERSION/bin/*
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-v0.15.5-beta.txt
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
```
```suggestion
  $ VERSION="0.16.3"
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/lnd-linux-arm64-$VERSION.tar.gz
```
```suggestion
  $ wget https://github.com/lightningnetwork/lnd/releases/download/v$VERSION-beta/lnd-linux-arm64-v$VERSION-beta.tar.gz
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.15.5-beta/manifest-roasbeef-v0.15.5-beta.sig.ots
+  $ VERSION="v0.15.5-beta"
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/lnd-linux-arm64-$VERSION.tar.gz
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-$VERSION.txt
```
```suggestion
  $ wget https://github.com/lightningnetwork/lnd/releases/download/v$VERSION-beta/manifest-v$VERSION-beta.txt
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
+  $ VERSION="v0.15.5-beta"
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/lnd-linux-arm64-$VERSION.tar.gz
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-$VERSION.txt
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-roasbeef-$VERSION.sig
```
```suggestion
  $ wget https://github.com/lightningnetwork/lnd/releases/download/v$VERSION-beta/manifest-roasbeef-v$VERSION-beta.sig
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/lnd-linux-arm64-$VERSION.tar.gz
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-$VERSION.txt
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-roasbeef-$VERSION.sig
+  $ wget https://github.com/lightningnetwork/lnd/releases/download/$VERSION/manifest-roasbeef-$VERSION.sig.ots
```
```suggestion
  $ wget https://github.com/lightningnetwork/lnd/releases/download/v$VERSION-beta/manifest-roasbeef-v$VERSION-beta.sig.ots
```

twofaktor commented on guide/lightning/lightning-client.md:
````diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
   ```sh
-  $ sha256sum --check manifest-v0.15.5-beta.txt --ignore-missing
-  > lnd-linux-arm64-v0.15.5-beta.tar.gz: OK
+  $ sha256sum --check manifest-$VERSION.txt --ignore-missing
````
```suggestion
  $ sha256sum --check manifest-v$VERSION-beta.txt --ignore-missing
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,19 +32,20 @@ We'll download, verify and install LND.
-  $ sha256sum --check manifest-v0.15.5-beta.txt --ignore-missing
-  > lnd-linux-arm64-v0.15.5-beta.tar.gz: OK
+  $ sha256sum --check manifest-$VERSION.txt --ignore-missing
+  > lnd-linux-arm64-$VERSION.tar.gz: OK
```
```suggestion
  > lnd-linux-arm64-v$VERSION-beta.tar.gz: OK
```

twofaktor commented on guide/lightning/lightning-client.md:
````diff
@@ -63,7 +64,7 @@ Now that we've verified the integrity of the downloaded binary, we need to check
 
   ```sh
-  $ gpg --verify manifest-roasbeef-v0.15.5-beta.sig manifest-v0.15.5-beta.txt
+  $ gpg --verify manifest-roasbeef-$VERSION.sig manifest-$VERSION.txt
````
```suggestion
  $ gpg --verify manifest-roasbeef-v$VERSION-beta.sig manifest-v$VERSION-beta.txt
```

twofaktor commented on guide/lightning/lightning-client.md:
````diff
@@ -77,7 +78,7 @@ We can also check that the manifest file was in existence around the time of the
 
   ```sh
-  $ ots verify manifest-roasbeef-v0.15.5-beta.sig.ots -f manifest-roasbeef-v0.15.5-beta.sig
+  $ ots verify manifest-roasbeef-$VERSION.sig.ots -f manifest-roasbeef-$VERSION.sig
````
```suggestion
  $ ots --no-cache verify manifest-roasbeef-v$VERSION-beta.sig.ots -f manifest-roasbeef-v$VERSION-beta.sig
```

twofaktor commented on guide/lightning/lightning-client.md:
````diff
@@ -91,8 +92,8 @@ Having verified the integrity and authenticity of the release binary, we can saf
   ```sh
-  $ tar -xzf lnd-linux-arm64-v0.15.5-beta.tar.gz
-  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lnd-linux-arm64-v0.15.5-beta/*
+  $ tar -xzf lnd-linux-arm64-$VERSION.tar.gz
````
```suggestion
  $ tar -xvf lnd-linux-arm64-v$VERSION-beta.tar.gz
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -91,8 +92,8 @@ Having verified the integrity and authenticity of the release binary, we can saf
-  $ tar -xzf lnd-linux-arm64-v0.15.5-beta.tar.gz
-  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lnd-linux-arm64-v0.15.5-beta/*
+  $ tar -xzf lnd-linux-arm64-$VERSION.tar.gz
+  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lnd-linux-arm64-$VERSION/*
```
```suggestion
  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lnd-linux-arm64-v$VERSION-beta/*
```

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -91,8 +92,8 @@ Having verified the integrity and authenticity of the release binary, we can saf
+  $ tar -xzf lnd-linux-arm64-$VERSION.tar.gz
+  $ sudo install -m 0755 -o root -g root -t /usr/local/bin lnd-linux-arm64-$VERSION/*
   $ lnd --version
   > lnd version 0.15.5-beta commit=v0.15.5-beta
```
```suggestion
  > lnd version $VERSION-beta commit=v$VERSION-beta
```

Swissnode commented:
@twofaktor thanks for the help, I think it was more useful for my Github skills than the changes themselves. You are right, was better to have just the numbers in the version and not the word "version". I checked your changes and they all checked out fine...

Think it's good to go - am ignoring the December changes as I think there are superceded

Swissnode commented:
I seem to have some "branch conflicts" but tbh i don't understand why it says this. Can you have a look?

twofaktor reviewed (approved):
tACK!

Mynima reviewed (changes requested):
Please update the follow lines of code in the "Future Update for Bitcoin Core" section.

Before this bit add ' VERSION = "25.0"
![image](https://github.com/raspibolt/raspibolt/assets/62206239/870834c4-59c8-45e8-918f-f863981be480)

Then correct the above 4 lines to use "$VERSION"

![image](https://github.com/raspibolt/raspibolt/assets/62206239/24af2439-e3a9-42fc-9ff2-74e0a420b060)

Then correct the above 2 lines to use "$VERSION"

Mynima commented:
> ok @Mynima and @twofaktor
> 
> I see it does pick up all the files I changed so I am now fairly sure I got the changes correctly made. Wanna review and pull them in?
> 
> Updated bitcoin-client.md and lightning-client.md

Some minor comments added. Thanks for all the work on this! Last push and we should get it across the line!!

twofaktor commented on guide/lightning/lightning-client.md:
```diff
@@ -32,18 +32,19 @@ We'll download, verify and install LND.
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.16.3-beta/manifest-v0.16.3-beta.txt
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.16.3-beta/manifest-roasbeef-v0.16.3-beta.sig
-  $ wget https://github.com/lightningnetwork/lnd/releases/download/v0.16.3-beta/manifest-roasbeef-v0.16.3-beta.sig.ots
+  $ VERSION="0.16.3"
```
```suggestion
  $ VERSION="0.16.3"
```
Add before this line:
$ VERSION="25.0"

@Swissnode

twofaktor reviewed (changes requested):
Add before this line: https://github.com/raspibolt/raspibolt/pull/1200/files#diff-1dad6164a6192352aaa1591ee62dc908f604c7df9cdae29a83ba0a0585b91539L528

`$ VERSION=25.0`

Mynima commented:
> Add instead of this line: https://github.com/raspibolt/raspibolt/pull/1200/files#diff-1dad6164a6192352aaa1591ee62dc908f604c7df9cdae29a83ba0a0585b91539L528
> 
> `$ VERSION=25.0`
> 
> @Swissnode, please

Both should work tbh

Mynima reviewed (approved):
tACK - LND

Bitcoin guide just needs a few bits added to the "Future Update for Bitcoin Core" section

twofaktor commented:
All ready, does someone want to do a final review?

Mynima reviewed (approved):
LGTM 
Bitcoin Core - utACK
LND - tACK

twofaktor reviewed (approved):
OK! All ready! Let's go to the master! 🚀

Swissnode commented:
exciting! it might be rather banal but it was my first ever PR, so I won't forget it! :D
